android/src/main/java/com/rncamerakit/CKCamera.kt

```diff
@@ -192,17 +195,28 @@ class CKCamera(context: ThemedReactContext) : FrameLayout(context), LifecycleObs
                 override fun onOrientationChanged(orientation: Int) {
                     val imageCapture = imageCapture ?: return
                     var newOrientation: Int = imageCapture.targetRotation
-                    if (orientation >= 315 || orientation < 45) {
-                        newOrientation = Surface.ROTATION_0
-                    } else if (orientation in 225..314) {
-                        newOrientation = Surface.ROTATION_90
-                    } else if (orientation in 135..224) {
-                        newOrientation = Surface.ROTATION_180
-                    } else if (orientation in 45..134) {
-                        newOrientation = Surface.ROTATION_270
+                    
+                    val deviceRotation = when {
+                        orientation >= 315 || orientation < 45 -> Surface.ROTATION_0
+                        orientation in 225..314 -> Surface.ROTATION_90
+                        orientation in 135..224 -> Surface.ROTATION_180
+                        orientation in 45..134 -> Surface.ROTATION_270
+                        else -> Surface.ROTATION_0
+                    }
+                    
+                    newOrientation = when (orientationMode) {
+                        "portrait" -> {
+                            Surface.ROTATION_0
+                        }
+                        "landscape" -> {
+                            Surface.ROTATION_90
+                        }
+                        else -> deviceRotation
                     }
```

Comment:
This seems to lose the value of `imageCapture.targetRotation` entirely. It'll always be changed by the when statement since it defaults to `deviceOrientation` which is always set to something else than `imageCapture.targetRotation`. I believe this would break the existing functionality.

Reply:
Good spot. I've rewritten this part.